### PR TITLE
Make ErrorResponse public

### PR DESCRIPTION
**Pull request by gperdomor**
This makes `ErrorResponse` internal and Decodable, useful for testing error responses xD 

### Checklist

- [x] Circle CI is passing (code compiles and passes tests).
- [x] There are no breaking changes to public API.
- [ ] New test cases have been added where appropriate.
- [ ] All new code has been commented with doc blocks `///`.

### CODE REVIEW

**Comment by gperdomor** [comment]
@tanner0101 can you merge this please :D

File: Sources/Vapor/Middleware/ErrorMiddleware.swift [hunk 1]
```diff
@@ -97,3 +88,12 @@ public final class ErrorMiddleware: Middleware, ServiceType {
         }
     }
 }
+
+/// Structure of `ErrorMiddleware` default response.
+public struct ErrorResponse: Encodable, Decodable {
```
Comment (rafiki270) [inline Sources/Vapor/Middleware/ErrorMiddleware.swift]: you just made this available throughout the whole Vapor, I think this should stay, although public, just within the `ErrorMiddleware` scope

Comment (gperdomor) [reply Sources/Vapor/Middleware/ErrorMiddleware.swift]: Like this: 

```swift
public final class ErrorMiddleware: Middleware, ServiceType {
    ...
    public struct ErrorResponse: Encodable, Decodable { ... }
    ...
}
```

Also works for me...

Comment (calebkleveter) [reply Sources/Vapor/Middleware/ErrorMiddleware.swift]: I think `ErrorMiddleware.Response` would be better.

Comment (rafiki270) [reply Sources/Vapor/Middleware/ErrorMiddleware.swift]: yeah (to both) :)

**Comment by rafiki270** [comment]
so sorry mate, I didn't realise this immediately, if this is for testing why not just `internal` and use `@testable import Vapor`?

**Comment by MrLotU** [comment]
Codewise looks good, but what’s the point of the PR?

File: Sources/Vapor/Middleware/ErrorMiddleware.swift [hunk 2]
```diff
@@ -1,5 +1,14 @@
 /// Captures all errors and transforms them into an internal server error HTTP response.
 public final class ErrorMiddleware: Middleware, ServiceType {
+    /// Structure of `ErrorMiddleware` default response.
+    internal struct ErrorResponse: Encodable, Decodable {
```
Comment (MrLotU) [inline Sources/Vapor/Middleware/ErrorMiddleware.swift]: I’d say make this inherit from `Codable` instead.

Comment (rafiki270) [reply Sources/Vapor/Middleware/ErrorMiddleware.swift]: hah, yeah! :D ... nice catch

Comment (rafiki270) [reply Sources/Vapor/Middleware/ErrorMiddleware.swift]: hah, yeah! :D ... nice catch

**Comment by gperdomor** [comment]
@rafiki270 done :D

**Comment by rafiki270** [comment]
@gperdomor yeah, I know dude, got the notification the same second my browser refreshed :D

**Comment by penny-coin** [comment]
Hey @gperdomor, you just merged a pull request, have a coin! 

You now have 2 coins.

**Pull request merged**